src/Symfony/Component/FeatureFlag/Debug/TraceableFeatureChecker.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag\Debug;
+
+use Symfony\Component\FeatureFlag\FeatureCheckerInterface;
+
+/**
+ * @experimental
+ */
+final class TraceableFeatureChecker implements FeatureCheckerInterface
+{
+    /** @var array<string, array{status: 'resolved'|'enabled'|'disabled', value: mixed, calls: int}> */
+    private array $resolvedValues = [];
+
+    public function __construct(
+        private readonly FeatureCheckerInterface $decorated,
+    ) {
+    }
+
+    public function isEnabled(string $featureName): bool
+    {
+        $isEnabled = $this->decorated->isEnabled($featureName);
```

Comment:
try / finally + new status : `error` ?

Reply:
Do you mean that this line should be put in a `try/catch`? It could be dangerous since exceptions will be silenced in dev but not in prod.

Reply:
not necessarily. It might be catched by the developper. But it won't know that this part failed. Still it adds more details / traces to understand where and when it failed. As you wish.